src/NServiceBus.Transport.RabbitMQ/Receiving/MessageConverter.cs

```diff
@@ -86,26 +91,26 @@ public static string DefaultMessageIdStrategy(BasicDeliverEventArgs message)
             return properties.MessageId;
         }
 
-        static Dictionary<string, string> DeserializeHeaders(IDictionary<string, object> headers)
+        static Dictionary<string, string> DeserializeHeaders(IDictionary<string, object> headers, int extraCapacity)
         {
-            var deserializedHeaders = new Dictionary<string, string>();
-
-            if (headers is Dictionary<string, object> messageHeaders)
```

Comment:
Technically it would also be possible to keep the pattern match while still doing these optimizations if we think it is worth it